client/lib/Hero.ts

```diff
@@ -497,8 +498,23 @@ class SessionConnection {
       createConnectionToCoreFn,
     );
 
+    if (connection !== connectionToCore) {
+      this.didCreateConnection = true;
```

Comment:
Isn't this more of a `didConnect`?

Reply:
I don't think so. It's not related to if connect actually gets called. More, did this hero create it's own "ConnectionToCore", which we need to clean up after